edge-modules/iotedge-diagnostics-dotnet/src/Program.cs

```diff
@@ -71,33 +77,57 @@ static async Task EdgeAgent(string managementUri)
 
-        static async Task Upstream(string hostname, string port, string proxy, string isNested)
+        static async Task LoadTrustBundle(string workload_uri)
+        {
+            string dummy = "0";
+
+            IEnumerable<X509Certificate2> trustBundle = await CertificateHelper.
+                GetTrustBundleFromEdgelet(new Uri(workload_uri), dummy, CLIENT_WORKLOAD_API_VERSION, dummy, dummy);
+            CertificateHelper.InstallCertificates(trustBundle, null);
+        }
+
+        static async Task Upstream(string hostname, string port, string proxy, string isNested, string workload_uri)
         {
+            bool nested = string.Equals(isNested, "true");
+
             if (port == "443")
             {
                 var httpClientHandler = new HttpClientHandler();
-                httpClientHandler.ServerCertificateCustomValidationCallback = (message, cert, chain, sslPolicyErrors) =>
-                {
-                    return true; // Is valid
-                };
 
                 if (proxy != null)
                 {
                     Environment.SetEnvironmentVariable("https_proxy", proxy);
                 }
 
+                if (nested)
+                {
+                    await LoadTrustBundle(workload_uri);
+                }
+
                 var httpClient = new HttpClient(httpClientHandler);
                 var logsUrl = string.Format("https://{0}/devices/0000/modules", hostname);
                 var httpRequest = new HttpRequestMessage(HttpMethod.Get, logsUrl);
-                HttpResponseMessage httpResponseMessage = await httpClient.SendAsync(httpRequest, HttpCompletionOption.ResponseHeadersRead);
-
-                if (isNested != null)
+                try
+                {
+                    HttpResponseMessage httpResponseMessage = await httpClient.SendAsync(httpRequest, HttpCompletionOption.ResponseHeadersRead);
+                    if (nested)
+                    {
+                        var keys = httpResponseMessage.Headers.GetValues("iothub-errorcode");
+                        if (!keys.Contains("InvalidProtocolVersion"))
+                        {
+                            throw new Exception($"Wrong value for iothub-errorcode header");
+                        }
+                    }
+                }
+                catch (Exception ex)
                 {
-                    var keys = httpResponseMessage.Headers.GetValues("iothub-errorcode");
-                    if (!keys.Contains("InvalidProtocolVersion"))
+                    string message = ex.Message;
+                    if ((ex.InnerException is AuthenticationException) && nested)
                     {
-                        throw new Exception($"Wrong value for iothub-errorcode header");
+                        message += "Make sure that the parent root certificate is part of this device trustbundle. Use the command 'openssl s_client -connect parent_hostname:443' to display parent certificate chain.";
```

Comment:
Hmm - appending this message to the exception message is interesting. How does the formatting of this look in the output? Can you paste the output?